File: DesignPatternsDotNet/StrategyPattern/Ship.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace StrategyPattern
{
    public class Ship
    {
        private string name;
        public IStrategyBehaviour work;
        public Ship(string aName)
        {
            name = aName;
            //Console.WriteLine("A ship is created");
        }
        public string Name
        {
            get
            {
                return name;
            }
        }
        public void GetToWork()
        {
            work.Carry();
        }
        public void SetToWork(IStrategyBehaviour aWork)
        {
            work = aWork;
        }
    }
}

File: DesignPatternsJava/src/com/strategy/Ship.java
package com.strategy;

public class Ship {
	private String name; 
	public IStrategyBehaviour work;
	public void getToWork(){
		work.carry();
	}
	public void setToWork(IStrategyBehaviour aWork){
		work=aWork;
	}
	public Ship(String aName){
		name=aName;
		//System.out.println("A ship is created");
	}
	public String getName() {
		return name;
	}		
}
